### PR TITLE
Added a one time initialization to the APP_INSTALLED event to record …

### PR DESCRIPTION
**Pull request by btkelly**
…when the application was first installed

### CODE REVIEW

File: amplify/src/main/java/com/github/stkent/amplify/tracking/GenericSettings.java [hunk 1]
```diff
@@ -79,4 +79,8 @@ public T getEventValue(@NonNull final IEvent event) {
         return null;
     }
 
+    public boolean hasEventValue(@NonNull final IEvent event) {
+        return sharedPreferences.contains(event.getTrackingKey());
```
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/GenericSettings.java]: This is potentially problematic if you track the same event in multiple ways e.g. with warmup and counter tracking... actually, the fact that the event tracking key is being used directly (without modification based on the type of the tracked value) seems problematic in and of itself?

File: amplify/src/main/java/com/github/stkent/amplify/tracking/IntegratedEvent.java [hunk 2]
```diff
@@ -20,12 +20,12 @@
 import android.support.annotation.NonNull;
 
 import com.github.stkent.amplify.tracking.interfaces.IEvent;
+import com.github.stkent.amplify.tracking.interfaces.ILogger;
 
 public enum IntegratedEvent implements IEvent {
 
     APP_CRASHED,
     APP_INSTALLED,
-    APP_UPDATED,
```
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/IntegratedEvent.java]: ?

File: amplify/src/main/java/com/github/stkent/amplify/tracking/GenericSettings.java (hunk 1 above)
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/GenericSettings.java]: Can add an abstract method to GenericSettings to request a prefix from each subclass?

File: amplify/src/main/java/com/github/stkent/amplify/tracking/IntegratedEvent.java (hunk 2 above)
Comment (btkelly) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/IntegratedEvent.java]: I didn't mean to take that out, it's not currently used so I took it out but realized someone else might want to use it.

File: amplify/src/main/java/com/github/stkent/amplify/tracking/GenericSettings.java (hunk 1 above)
Comment (btkelly) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/GenericSettings.java]: Hmm that was one of the things my refactor removed... Let me think on this one and how to get that back. Any ideas how we should adjust this?

File: amplify/src/main/java/com/github/stkent/amplify/tracking/TrackedEvent.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    private final IEvent event;
+    private final IEventCheck eventCheck;
+
+    public TrackedEvent(@NonNull final IEvent event, @NonNull final IEventCheck eventCheck) {
+        this.event = event;
+        this.eventCheck = eventCheck;
+    }
+
+    public String getTrackingKey() {
+        return "AMPLIFY_" + event.getTrackingKey() + "_" + eventCheck.getTrackingKey();
```
Comment (stkent) [inline amplify/src/main/java/com/github/stkent/amplify/tracking/TrackedEvent.java]: :+1:

**Pull request merged**